### ISSUE TITLE
Emails Blocked By Server

### ISSUE THREAD

**Issue by asithade** [atech/postal#146]
I'm not sure if this is where I can get some help on this but I'm pretty stuck at the moment. 

I've setup my DKIM, SPF and DMARC records but for some reason, the only server accepting my emails are Gmail servers.

When I send to any Microsoft Live accounts (Outlook, Hotmail, Live), it says that it has been sent and accepted but nothing shows up on the other end. Not even in Junk Mail.

However, unlike any emails sent from gmail which return a 250 response, and says: `250 2.0.0 OK 1494661174 y193si5439120pgd.41 - gsmtp`, anything sent to a Microsoft Live account says `250 <70c4cf3f-6ad9-4b9a-b623-3a4aca4bbf25@rp.smtp.domain.org> Queued mail for delivery`

When I send to any Apple servers, it gets rejected with the error: 
`553 5.1.8 Domain of sender address does not exist`

There are A records pointing to my server so I don't see what would cause that. 

I've done the mail-tester.com test and got a 10/10.

I'm wondering if it's because I'm using AWS EC2 and it's being blocked by providers?

**Comment by mattmartincouk** [comment]
Domain of sender address does not exist is probably the biggest clue.

What is the sending domain?

**Comment by asithade** [comment]
@mattmartincouk it's rallystarter.org

**Comment by mattmartincouk** [comment]
Can you output the full headers from a message that is getting through?

**Comment by asithade** [comment]
```
received
from web-ui (ip-119-228-239-173.router.siliconvalleywifi.com [173.239.228.119]) by Postal with HTTP; Sat, 13 May 2017 23:40:43 +0000
date
Sat, 13 May 2017 23:40:43 +0000
from
test@rallystarter.org
to
asithade@hotmail.com
message-id
<e3c8b574-ed91-4197-b7c6-28f1a6099aa8@rp.smtp.rallystarter.org>
subject
Test Message at May 13, 2017 23:40
mime-version
1.0
content-type
text/plain; charset=UTF-8
content-transfer-encoding
7bit
dkim-signature
v=1; a=rsa-sha256; c=relaxed/relaxed; d=rallystarter.org; s=postal-dd8GOi; t=1494718843; bh=mvuu3b9qEfdaZCSaVxLvRSqKbMB5YN4qxRCRw5FtvuE=; h=date:from:to:message-id:subject; b=FNp1+6pSICA9PiJmfTRjsQB7RpCcAqHodsqdW9+ZmwOSTiaHpg9H6MAKg9Wf9//hC0wYGubq7cbz8cdTA8IPXJU51FKHCXfekSeYLrQj/yuD7fULaUaHiDLjHtT3uRJWGT2ghXl4JzXqS8czma7ucauT8VC50o3G9dz4DgKh08A=
x-postal-msgid
ewaLiV6BRQ1B
```

**Comment by mattmartincouk** [comment]
Is that the headers from the GUI or the receiving end? Need them from the receiving end from any of the mails that got through.

**Comment by asithade** [comment]
That was from GUI.

```
Delivered-To: asithade@gmail.com
Received: by 10.12.175.210 with SMTP id t18csp555756qvc;
        Sat, 13 May 2017 01:14:37 -0700 (PDT)
X-Received: by 10.98.130.1 with SMTP id w1mr8659523pfd.128.1494663277395;
        Sat, 13 May 2017 01:14:37 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1494663277; cv=none;
        d=google.com; s=arc-20160816;
        b=y0nt+bX/YottYwdX4Qy5TAjFZg6oh9iqmkIPvu0DIvmTzGNkZQizJ+QFbHqVv6vOLE
         rLviAasy7IU22gsJfbHFNkQc4B3ak7vw2Pic4hisVA6+my9/X6RkP4x3qGFVQQPVMZxi
         87Oqd1ZxikVMWrvETd8qsABO3asepcDKukLaBooZBmZKJMJTeqgDHVGydi0FnI7VGAtg
         vry1GhoOuTFO9cwrQx0p4c6lxWVuayDyLfx95G/bDEoWyR27wl/pXvJR7EfvmlN/LvCv
         rDLpdNfk6arap7SEUyKNumnNIDQORJ76XfwB8Kkg+eMrHXZHFTd65fkkNzl7r+hrM+Xk
         lF4w==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=content-transfer-encoding:mime-version:subject:message-id:to:from
         :date:dkim-signature:resent-sender:arc-authentication-results;
        bh=mvuu3b9qEfdaZCSaVxLvRSqKbMB5YN4qxRCRw5FtvuE=;
        b=f89I5vsMmwKnKmKB9v1K2svgEhwHsTHwIAt8niOGCpJvX2vO7Tg4EBlZYnWR298Rgc
         yeHCzffpX2So1HNRNenHlOjwtprYvKB6ycdyEchMem+GYHJLfMF4WWY0aE0d55baOvVH
         P+jgjuiXA8/UJ/DTNfzBnQRe1Ijbds8ue1QbIBd001U+HyrxMs09VJONtW+ddLAegoUp
         EnBDmDD3qS73N46P0m3vsjngb8la12Dx0agn90bk2OZFEL+TVe2I1E41edNEBQQDOHPN
         mTiBhFk1e6cXjKKG7yeMpLQ+G/vqwnmbU94WGgsCA/WjAvqaQS+wWu6yDMGk++EAwAGs
         T3zg==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=pass header.i=@rallystarter.org;
       spf=pass (google.com: domain of inqo4d@psrp.rallystarter.org designates 34.210.98.242 as permitted sender) smtp.mailfrom=inqo4d@psrp.rallystarter.org;
       dmarc=pass (p=NONE sp=NONE dis=NONE) header.from=rallystarter.org
Return-Path: <inqo4d@psrp.rallystarter.org>
Received: from smpt.rallystarter.org (postal.rallystarter.org. [34.210.98.242])
        by mx.google.com with ESMTPS id 59si5392509pld.76.2017.05.13.01.14.36
        for <asithade@gmail.com>
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Sat, 13 May 2017 01:14:37 -0700 (PDT)
Received-SPF: pass (google.com: domain of inqo4d@psrp.rallystarter.org designates 34.210.98.242 as permitted sender) client-ip=34.210.98.242;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@rallystarter.org;
       spf=pass (google.com: domain of inqo4d@psrp.rallystarter.org designates 34.210.98.242 as permitted sender) smtp.mailfrom=inqo4d@psrp.rallystarter.org;
       dmarc=pass (p=NONE sp=NONE dis=NONE) header.from=rallystarter.org
Resent-Sender: inqo4d@psrp.rallystarter.org
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=rallystarter.org; s=postal-dd8GOi; t=1494663276; bh=mvuu3b9qEfdaZCSaVxLvRSqKbMB5YN4qxRCRw5FtvuE=; h=date:from:to:message-id:subject; b=jf8gfk53h6mEHWPsaeoj5DZt0F+u2fVck6h30cRkv5tifvqhGmkp4rEGG+plNph282bz/z/M05RLddio7HcybJ78v//0kpiVHj7pL2ctJaOUJ7P4RMQfiaDZ3Z/PJRnOVZs5FmZnhF+D3FyVb1S74V0U3zRi5/EzoSgTOxmyYgc=
X-Postal-MsgID: HSVe62eD57Rv
Received: from web-ui (c-67-169-16-68.hsd1.ca.comcast.net [67.169.16.68]) by Postal with HTTP; Sat, 13 May 2017 08:14:36 +0000
Date: Sat, 13 May 2017 08:14:36 +0000
From: test@rallystarter.org
To: asithade@gmail.com
Message-ID: <986c4c19-b3ee-453a-ac30-1747d6fda394@rp.smtp.rallystarter.org>
Subject: Test Message at May 13, 2017 08:14
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit

This is a message to test the delivery of messages through Postal.
```

**Comment by mattmartincouk** [comment]
DNS is your issue.

psrp.rallystarter.org points to rp.smtp.rallystarter.org
rp.smtp.rallystarter.org does not exist

Review: https://github.com/atech/postal/wiki/Domains-&-DNS-Configuration

**Comment by asithade** [comment]
Thank you for that! Looks like that fixed the issue with Apple.com, but I guess Microsoft Live accounts are still rejecting my emails unless I add the sender as a contact.

**Comment by mattmartincouk** [comment]
Maybe - could be propagation too.

If sorted, would close this issue as it is more config than software.

**Comment by adamcooke** [comment]
Some providers don't like new IPs regardless of how well DNS is configured. They do eventually sort themselves out. You might be able to help by configuring feedback loops with them.

**Comment by maks200179** [comment]
Hi there . 
I have almost same issue  and stack .
Please help me . 

I have configured my DKIM, SPF and DMARC records correctly.
I think that i know what cause the problem but i don't know how to resolve it . 
Thank for future advice . 
There are A records pointing to my server so using telnet everything look fine . 


Gmail headers of email sanded from postal : 

Delivered-To: maxfrXXXX@gmail.com
Received: by 2002:ab0:1548:0:0:0:0:0 with SMTP id p8csp625817uae;
        Wed, 3 Feb 2021 15:13:59 -0800 (PST)
X-Google-Smtp-Source: ABdhPJw5xROdFM+MG/MuUs035NkBNRGvoIvQSU+AmGVjEf1MyIlmzfsEfI0wEQABpAEJBO12j+Uz
X-Received: by 2002:a17:906:e106:: with SMTP id gj6mr5304785ejb.337.1612394039050;
        Wed, 03 Feb 2021 15:13:59 -0800 (PST)
ARC-Seal: i=1; a=rsa-sha256; t=1612394039; cv=none;
        d=google.com; s=arc-20160816;
        b=ielo/RLHla/QlMbDEqOgCPygaXyPsyl31lJvtYHdrbGuXBqeecPgFX2Cg+l8/2XPRj
         zKKQu2BJkgI06FGq+nDNg+5v7bWE3pTEsDI/L25uTKFsr9Jrcl++E6+GW4UrJQLvAxSy
         +0Orki9P/xUls2HqaGOU36QsN96UQYkz6AWrfZzN+5E8t7Ntgg0UuvHEc5MSpIMO1rpb
         Sj/x/Yh3wY4WWl/TPd2XfI1HXv6IvT3OBpLCvtG0P6Zmi+qdHkDeDhb6vk4rNUMvcw76
         nluNSO9VSjv83rbFnM/gq3swBsG7c9KW+U6poDdK+DJzYUpRg9CGs4Lced4MUpryA//S
         FyNA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=content-transfer-encoding:mime-version:subject:message-id:to:from
         :date:dkim-signature:resent-sender;
        bh=RwML7bZTV0QEdoeJq45vt7A1976/m7ol7MMBjBCLcIM=;
        b=LheGMxSOSarkVySKY1DwpRtfzeamis3j1+Xd5Xw3mmjqoXlXfzIngxrhAxS9MWNBxR
         IfHIt9TL/kFqSjv+lJ7JQkM1i7F8jvpTllFMxVpjhIWdBPUu1r54V3NQgQL47TyJDMVR
         NFrjuo6AFpJrEW2RDHZyIn5IPmEyBdq30gqfzBqoiuWH8ofTit0r4a0BYCzwWMy81J/J
         p7JcozKgHSKy4xfI9nNfUtr8uI+WRP3CtHdC6tqSiOzQ0syd4L93+vOBp6Q4t1+k76YG
         OzgO1itpTsjeFr7ZsmU3L3eiCKhWiTa+k+ZsQvtlsic5eZ025l/Bz+e0dq1J06168WGR
         5YNA==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=neutral (body hash did not verify) header.i=@xxxmaxfr.com header.s=postal-Adzx8X header.b=hycIOfA1;
       spf=neutral (google.com: 84.228.160.xxx is neither permitted nor denied by best guess record for domain of cughzi@psrp.xxxmaxfr.com) smtp.mailfrom=cughzi@psrp.xxxmaxfr.com
Return-Path: <cughzi@psrp.xxxmaxfr.com>
Received: from mail.xmaxfr.com (IGLD-84-228-160-102.inter.net.il. [84.228.160.xxx])
        by mx.google.com with ESMTPS id z17si2320314edl.91.2021.02.03.15.13.58
        for <maxfrXXXX@gmail.com>
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 03 Feb 2021 15:13:59 -0800 (PST)
Received-SPF: neutral (google.com: 84.228.160.xxx is neither permitted nor denied by best guess record for domain of cughzi@psrp.xxxmaxfr.com) client-ip=84.228.160.xxx;
Authentication-Results: mx.google.com;
       dkim=neutral (body hash did not verify) header.i=@xxxmaxfr.com header.s=postal-Adzx8X header.b=hycIOfA1;
       spf=neutral (google.com: 84.228.160.xxx is neither permitted nor denied by best guess record for domain of cughzi@psrp.xxxmaxfr.com) smtp.mailfrom=cughzi@psrp.xxxmaxfr.com
Resent-Sender: cughzi@psrp.xxxmaxfr.com
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=xxxmaxfr.com; s=postal-Adzx8X; t=1612394037; bh=+ESn9ALTRi6ARpK6byMSwGuOWq1ov3AMtZD/HY4DxAE=; h=date:from:to:message-id:subject:mime-version:content-type:content-transfer-encoding; b=hycIOfA1O90MRzwriR90c+BM9V+A1GCKLCI7/JpAoB9M2V1yXGZzqsKp3zcQeqhULKutPJusZ4Ge0sdtPKSnFlJHRSZBmCoxTl9chWkASvGn1rNzGiNFwfauKJxScZx+Rg+he67U8tBR6U6DsPt8CpiJnZkfaItqjI79PzDHTEs=
X-Postal-MsgID: jh7r7E4B379S
Received: from web-ui (proxyserver.home-test-email-validator_infrastructure [172.22.0.3]) by Postal with HTTP; Wed, 03 Feb 2021 23:13:57 +0000
Date: Wed, 03 Feb 2021 23:13:57 +0000
From: maksim123@xxxmaxfr.com
To: maxfrXXXX@gmail.com
Message-ID: <fb777479-79ff-4f29-ba57-604f2e7138d7@rp.mail.xxxmaxfr.com>
Subject: Test Message at February 03, 2021 23:13
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit


This line come from my revers proxy docker .  
Received: from web-ui (proxyserver.home-test-email-validator_infrastructure [172.22.0.3]) by Postal with HTTP; Wed, 03 Feb 2021 
But i use smtp as localhost 25 of postal inside docker, and map port  to host machine like this.
I really do not understand how is it work.






    postal:
        #image: catdeployed/postal:alpine
        build:
            context: postal/
            dockerfile: Dockerfile
        container_name: postal
        privileged: true  
        command: run
        ports:
          - 0.0.0.0:25:25
        depends_on:
          - "mysql"
          - "rabbitmq"
        volumes:
          - ./postal/data/persistent_assets:/storage
          
          - ./postal/data/config/lets_encrypt.pem:/opt/postal/config/lets_encrypt.pem
          - ./postal/data/config/signing.key:/opt/postal/config/signing.key
          #- ./postal/data/postal_assets:/opt/postal/public/assets
          - ./postal/src/templates/:/templates
        environment:
          - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
          - MYSQL_DATABASE=postal
          - RABBITMQ_DEFAULT_USER=postal
          - RABBITMQ_DEFAULT_PASS=changeme
          - RABBITMQ_DEFAULT_VHOST=postal
          - POSTAL_FNAME=Sia
          - POSTAL_LNAME=Cat
          - POSTAL_PASSWORD=StrongPassword
          - POSTAL_EMAIL=maximov@xxxmaxfr.com
        healthcheck:
          test: curl -sS --resolve http://postal.home.test:5000:127.0.0.1 http://postal.home.test:5000 || exit 1
          interval: 5s
          timeout: 10s
          retries: 3
        networks:
            infrastructure:
                aliases:
                    - postal.home.test







    proxyserver:
        container_name: proxyserver
        build:
            context: ./proxyserver/
            dockerfile: Dockerfile
        #volumes:
            #- ./proxyserver/config/certs:/etc/ssl/certs
            #- ./proxyserver/config/private:/etc/ssl/private 
        ports:
            - "80:80"
            - "443:443"
            - "42:22"
        networks:
            - infrastructure
        tty: true
        
        depends_on:
            - postal
        
       
            

                  

            
networks:
    infrastructure:
        driver: bridge


 i have mounted this files  because postal fail after reboot on error about cant find /opt/postal/config/lets_encrypt.pem key .
          volumes:
              - ./postal/data/config/lets_encrypt.pem:/opt/postal/config/lets_encrypt.pem
              - ./postal/data/config/signing.key:/opt/postal/config/signing.key





  SMTP Banner Check  Reverse DNS does not match SMTP Banner 

  SMTP TLS | Warning - Does not support TLS. 
  SMTP Transaction Time | 8.564 seconds - Not good! on Transaction Time
  SMTP Reverse DNS Mismatch  OK - 84.228.160.XXX resolves to IGLD-84-228-160-XXX.inter.net.il   
  SMTP Valid Hostname  OK - Reverse DNS is a valid Hostname   
  SMTP Connection Time  0.617 seconds - Good on Connection time  
  SMTP Open Relay  OK - Not an open relay.   

Connecting to 84.228.160.XXX

220 mail.xmaxfr.com ESMTP Postal/BNVK2P [464 ms]
EHLO keeper-us-east-1c.mxtoolbox.com
250-My capabilities are
250 AUTH CRAM-MD5 PLAIN LOGIN [726 ms]
MAIL FROM:<supertool@mxtoolboxsmtpdiag.com>
250 OK [260 ms]
RCPT TO:<test@mxtoolboxsmtpdiag.com>
530 Authentication required [6813 ms]

LookupServer 9390ms

**Comment by willpower232** [comment]
@maks200179 please open a new issue rather than commenting on long-closed issues
